compiler/src/dotty/tools/repl/ScriptEngine.scala

```diff
@@ -0,0 +1,90 @@
+package dotty.tools
+package repl
+
+import java.io.{Reader, StringWriter};
+import javax.script.{AbstractScriptEngine, Bindings, ScriptContext, ScriptEngine => JScriptEngine, ScriptEngineFactory, ScriptException, SimpleBindings}
+
+/** A JSR 223 (Scripting API) compatible wrapper around the REPL for improved
+ *  interoperability with software that supports it.
+ *
+ *  It works by instantiating a new script engine through the script engine manager.
+ *  The script engine provides a eval method to evaluate scripts in string form.
+ *  Example use:
+ *
+ *  val m = new javax.script.ScriptEngineManager()
+ *  val e = m.getEngineByName("scala")
+ *  println(e.eval("42"))
+ *
+ *  @param factory the script engine factory
+ */
+class ScriptEngine(factory: ScriptEngine.Factory) extends AbstractScriptEngine {
```

Comment:
Does it need to take a factory?
```scala
class ScriptEngine extends AbstractScriptEngine {
  ...
  private[this] val driver = new ReplDriver(Array("-usejavacp", "-color:never"), Console.out, None)
  ...
  def getFactory: ScriptEngineFactory = new ScriptEngineFactory {
    ...
    def getScriptEngine = new ScriptEngine
  }
```

Reply:
Well, it avoids to re-instantiate one each time the method is called..

Reply:
Well, it allows to reuse the same driver in all new script engines..

Reply:
If you share the same driver, you will share the same classloader across multiple REPL sessions. Is it intended?

Reply:
I have nothing against it.

Reply:
To me, this is asking for trouble. If you run two script engines in parallel, they will override each other output and you will get `ClassNotFoundException`...

Unless there is a good reason to share the driver, I would argue against it.

Reply:
ok, done.